interpretation/deepseismic_interpretation/azureml_pipelines/base_pipeline.py

```diff
@@ -0,0 +1,387 @@
+from azureml.core.runconfig import DEFAULT_GPU_IMAGE
+from azureml.core.conda_dependencies import CondaDependencies
+from msrest.exceptions import HttpOperationError
+from azureml.data.data_reference import DataReference
+from azureml.core import Environment
+from dotenv import load_dotenv
+import os
+import re
+from abc import ABC, abstractmethod
+import json
+
+
+class DeepSeismicAzMLPipeline(ABC):
+    """
+    Abstract base class for pipelines in AzureML
+    """
+    def __init__(self, pipeline_config, ws_config=None):
+        """
+        constructor for DeepSeismicAzMLPipeline class
+
+        :param str pipeline_config: [required] path to the pipeline config file
+        :param str ws_config: [optional] if not specified, will look for
+                              .azureml/config.json. If you have multiple config files, you
+                              can specify which workspace you want to use by passing the
+                              relative path to the config file in this constructor.
+        """
+        self.ws = Workspace.from_config(path=ws_config)
+        self._load_environment()
+        self._load_config(pipeline_config)
+        self.steps = []
+        self.pipeline_tags = None
+        self.last_output_data = None
+
+    def _load_config(self, config_path):
+        """
+        helper function for loading in pipeline config file.
+
+        :param str config_path: path to the pipeline config file
+        """
+        try:
+            with open(config_path, 'r') as f:
+                self.config = json.load(f)
+        except Exception as e:
+            raise Exception("Was unable to load pipeline config file. {}".format(e))
+
+    @abstractmethod
+    def construct_pipeline(self):
+        """
+        abstract method for constructing a pipeline. Must be implemented by classes
+        that inherit from this base class.
+        """
+        raise NotImplementedError("construct_pipeline is not implemented")
+
+    @abstractmethod
+    def _setup_steps(self):
+        """
+        abstract method for setting up pipeline steps. Must be implemented by classes
+        that inherit from this base class.
+        """
+        raise NotImplementedError("setup_steps is not implemented")
+
+    def _load_environment(self):
+        """
+        loads environment variables needed for the pipeline.
+        """
+        load_dotenv()
+        self.account_name = os.getenv("BLOB_ACCOUNT_NAME")
+        self.container_name = os.getenv("BLOB_CONTAINER_NAME")
+        self.account_key = os.getenv("BLOB_ACCOUNT_KEY")
+        self.blob_sub_id = os.getenv("BLOB_SUB_ID")
+
+        self.comp_name = os.environ.get("AML_COMPUTE_CLUSTER_NAME")
+        self.comp_min_nodes = os.environ.get("AML_COMPUTE_CLUSTER_MIN_NODES")
+        self.comp_max_nodes = os.environ.get("AML_COMPUTE_CLUSTER_MAX_NODES")
+        self.comp_vm_size = os.environ.get("AML_COMPUTE_CLUSTER_SKU")
+
```

Comment:
@maxkazmsft @annazietlow This is where the .env file is loaded. Max do you want to see this changed or is adding additional documentation to the README.md for users to create a .env file sufficient?

Reply:
That looks fine - adding documentation wouldn't hurt if you think it's needed

Reply:
I will add some documentation in the README.md :)